### PR TITLE
Add HED tags to eeg_matchingpennies dataset

### PR DESCRIPTION
**Pull request by sappelhoff**
The eeg_matchingpennies dataset cannot be very well re-used for several reasons:

- there are only 10 EEG electrodes recorded
- very few events were recorded
- ...

**However** it can still serve as a valuable example to showcase some BIDS principles.

That's why I want to add HED tags to it. I took a draft from @VisLab that she sent to me via email some months ago.

To use an HED tagging strategy, I re-formatted the dataset to some extend ... so that it uses a "trial based" events.tsv file again --> each row in events.tsv represents a trial, which contains several events.

What I am stuck at is how to format the brackets in the HED tags. This is the relevant part, I added linebreaks into the HED strings for readability:

```json
"trial_type": {
    "Description": "Which hand was raised, and did this match the BCIs prediction (lose outcome), or not (win outcome).",
    "Levels": {
        "raised-left/match-false": "raised left hand, bci predicted right hand",
        "raised-left/match-true": "raised left hand, bci predicted left hand",
        "raised-right/match-false": "raised right hand, bci predicted left hand",
        "raised-right/match-true": "raised right hand, bci predicted right hand"
    },
    "HED": {
        "raised-left/match-true": "Description/Subject raised left hand and computer presented image of left hand,
(Agent-action, Experiment-subject, (Lift, (Left-side, Hand), (Release, (Left-side, Button)), ((Delay/68 ms, Approximated),
(Sensory-presentation, Visual, (Drawing, ID/left_hand.png, (Left-side, Hand), (Feedback, (Intended-effect, Penalty)))",

        "raised-left/match-false": "Description/Subject raised left hand and computer presented image of right hand,
(Agent-action, Experiment-subject, (Lift, (Left-side, Hand), (Release, (Left-side, Button)), ((Delay/68 ms, Approximated),
(Sensory-presentation, Visual, (Drawing, ID/right_hand.png, (Right-side, Hand), (Feedback, (Intended-effect, Reward)))",

        "raised-right/match-true": "Description/Subject raised right hand and computer presented image of right hand,
(Agent-action, Experiment-subject, (Lift, (Right-side, Hand), (Release, (Right-side, Button)), ((Delay/68 ms, Approximated),
(Sensory-presentation, Visual, (Drawing, ID/right_hand.png, (Right-side, Hand), (Feedback, (Intended-effect, Penalty)))",

        "raised-right/match-false": "Description/Subject raised right hand and computer presented image of left hand,
(Agent-action, Experiment-subject, (Lift, (Right-side, Hand), (Release, (Right-side, Button)), ((Delay/68 ms, Approximated),
(Sensory-presentation, Visual, (Drawing, ID/left_hand.png, (Left-side, Hand), (Feedback, (Intended-effect, Reward)))"
    }
}
```

The HED validator rightfully raises an issue that more brackets are opened than are closed. I wanted to fix this, but I am not really sure where to start learning about how to do this. Can you help @VisLab?

### CODE REVIEW

**Review by sappelhoff — COMMENTED** [review]
I managed to figure out the brackets (I think). But now the HED validator tells me that I used invalid tags :-)

I'll look into it again at some other point.

**Comment by VisLab** [comment]
I am hoping that your approval of bids-validator PR #1313 will solve the
validation issue.  However, you should really wait and tag with HED
8.0.0-Beta.5 which is only on hedxml-test right now but will hopefully be
released as the official HED8.0.0 within the week.

File: eeg_matchingpennies/task-matchingpennies_events.json [hunk 1]
```diff
@@ -29,38 +25,62 @@
...
+            "raised-left/match-false": "Description/Subject raised left hand and computer presented image of right hand, (Agent-action, Experiment-subject, (Lift, (Left-side, Hand)), (Release, (Left-side, Button)), ((Delay/68 ms, Approximated), (Sensory-presentation, Visual), (Drawing, ID/right_hand.png, (Right-side, Hand)), (Feedback, (Intended-effect, Reward))))",
+            "raised-right/match-true": "Description/Subject raised right hand and computer presented image of right hand, (Agent-action, Experiment-subject, (Lift, (Right-side, Hand)), (Release, (Right-side, Button)), ((Delay/68 ms, Approximated), (Sensory-presentation, Visual), (Drawing, ID/right_hand.png, (Right-side, Hand)), (Feedback, (Intended-effect, Penalty))))",
+            "raised-right/match-false": "Description/Subject raised right hand and computer presented image of left hand, (Agent-action, Experiment-subject, (Lift, (Right-side, Hand)), (Release, (Right-side, Button)), ((Delay/68 ms, Approximated), (Sensory-presentation, Visual), (Drawing, ID/left_hand.png, (Left-side, Hand)), (Feedback, (Intended-effect, Reward))))"
+        }
+    },
```
Comment (sappelhoff) [inline eeg_matchingpennies/task-matchingpennies_events.json:63]: @VisLab can you help me finding the correct HED vocabulary for what I want to express here? These tags are based on a draft you sent me several months ago - and now that HED 8.0.0 is released (:rocket:), the HED validator is telling me that some of these tags are not valid.

See also the report here: https://github.com/bids-standard/bids-examples/pull/265/checks?check_run_id=3280815398#step:8:1584

and copy paste of the first lines:

```
		./task-matchingpennies_events.json
			Evidence: ERROR: [HED_TAG_INVALID] Invalid tag - "Experiment-subject"
		./task-matchingpennies_events.json
			Evidence: ERROR: [HED_TAG_INVALID] Invalid tag - "Left-side"
		./task-matchingpennies_events.json
			Evidence: ERROR: [HED_TAG_INVALID] Invalid tag - "Left-side"
		./task-matchingpennies_events.json
			Evidence: ERROR: [HED_TAG_INVALID] Invalid tag - "Button"
		./task-matchingpennies_events.json
			Evidence: ERROR: [HED_TAG_INVALID] Invalid tag - "Approximated"
		./task-matchingpennies_events.json
			Evidence: ERROR: [HED_TAG_INVALID] Invalid tag - "Visual"
		./task-matchingpennies_events.json
			Evidence: ERROR: [HED_TAG_INVALID] Invalid tag - "Left-side"
		./task-matchingpennies_events.json
			Evidence: ERROR: [HED_TAG_INVALID] Invalid tag - "Delay/68 ms, Appro"
		./task-matchingpennies_events.json
			Evidence: ERROR: [HED_TAG_INVALID] Invalid tag - "(Sensory-"
		./task-matchingpennies_events.json
			Evidence: ERROR: [HED_TAG_INVALID] Invalid tag - ", (Drawin"
```

**Comment by sappelhoff** [comment]
Okay, this is what I come up with, using https://www.hedtags.org/display_hed.html?version=8.0.0

This tags the event `"raised-left/match-true"`, where the following three things happened:

1. a participant lifted their left ring finger from a mouse button of a mouse placed to their left
2. a short delay occurred (approximately 68ms)
3. a drawing of a hand was visually presented on the left side of a computer screen, symbolizing a penalty

```
(Agent-action,
 Participant-reponse,
    (Lift,
        (Ring-finger,
            (Left-side-of, Experiment-participant)
        )
    ),
    (Release,
        (Mouse-button,
            (Left-side-of, Experiment-participant)
        )
    )
),

(Delay,
 Duration,
     (Approximately-equal-to, 68ms)
),

(Sensory-event,
 Visual-presentation,
     (Drawing,
         (ID/left_hand.png, Hand, (Left-side-of, Computer-screen))
     ),
     (Feedback, Penalty)
)
```

Does this make sense @VisLab ?

File: eeg_matchingpennies/task-matchingpennies_events.json (hunk 1 above)
Comment (sappelhoff) [reply eeg_matchingpennies/task-matchingpennies_events.json]: I took some time this morning and I think it's fine now, see https://github.com/bids-standard/bids-examples/pull/265#issuecomment-1013855511

**Comment by VisLab** [comment]
Sorry, I am having trouble following. The experiment describes lifting the left or right hand. I am not sure where the ring finger and mouse button are coming from.  Maybe we could we start with good text descriptions of each event?  Thanks...

**Comment by sappelhoff** [comment]
> Sorry, I am having trouble following.... The experiment describes lifting the left or right hand. I am not sure where the ring finger and mouse button are coming from. Maybe we could we start with good text descriptions of each event? Thanks...

True ... I will expand on this in the README. Participants were sitting in front of the screen, one computer mouse glued to the table to their right, and another computer mouse glued to the table to their left. The participants were then instructed to place their hands on the table, the respective ring finger of each hand on a button of the respective mouse (left/right). Then during the experiment, participants were instructed to raise hands ... I should correct that HED from Ring-finger to Hand.

The only reason they were instructed to press the mice buttons with their ring fingers was, that these fingers are more effortful to lift individually (as opposed to e.g., index fingers), and so this encourages participants to properly lift their hand, instead of just a finger.

> Also, the original event organization was by trial. I was thinking maybe we need to unfold the events for effective tagging.

Yes, we had talked about this a lot in the past - and I actually tried to unfold the events ... but it turned out that I had way too little information to effectively do that. That's why in this PR, I am reverting to the original "by trial" organization.

> This might make a good basis for a tutorial in what the impact is of the different organizations on downstream analysis.

I think for a tutorial we should take a dataset where not as much information has been lost as for this dataset (regrettably)

**Pull request merged**